brave/brave-core
# feat(wallet): ZCash Text Memo

## Description 

Adds a `Memo` field on the `Send` screen for `ZCash` Shielded transactions.


Resolves <https://github.com/brave/brave-browser/issues/41986>
Resolves <https://github.com/brave/brave-browser/issues/42078>



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1. Make sure that `ZCash Shielded Transactions` is enabled
2. Open the `Wallet` and navigate to the `Send` tab
3. Select `ZCash` as the `Token` to send.
4. Paste in a ZCash `Unified Shielded` address to send to
5. You should now see a `Add memo` button, click it
6. Enter a memo `Message` and create the transaction
7. Check that the `Memo` is displayed in the `Confirm Transaction` panel
8. Check that the `Memo` is displayed in the `Activity` tab.

https://github.com/user-attachments/assets/e7e87214-8e8d-4ce8-a6f4-3b658ed606c6
